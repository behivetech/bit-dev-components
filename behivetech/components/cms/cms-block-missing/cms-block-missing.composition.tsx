import * as React from 'react';
import { CmsBlockMissing } from './cms-block-missing';

export const BasicCmsBlockMissing    = () => (
    <CmsBlockMissing componentName="mockMissingComponent" role="button" />
);
