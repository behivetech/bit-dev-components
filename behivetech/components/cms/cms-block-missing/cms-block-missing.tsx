import * as React from 'react';

// Extending the HTMLAttributes for HTMLDivElement so all attributes of div can be passed in.
export interface CmsBlockMissingProps extends React.HTMLAttributes<HTMLDivElement> {
    /**
     * name of component that was missing
     */
    componentName: string,
}

const STYLES = {
    backgroundColor: '#900',
    color: '#fff',
};

export function CmsBlockMissing({componentName, style = STYLES, ...restProps}: CmsBlockMissingProps) {
    return (
        <div className="cms-block-missing" style={style} {...restProps}>
            COMPONENT IMPORT ERROR: The {componentName} Componpent is not available from the importedBlocks prop.
        </div>
    );
}

