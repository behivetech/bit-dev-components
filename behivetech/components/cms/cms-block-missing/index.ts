export { CmsBlockMissing } from './cms-block-missing';
export type { CmsBlockMissingProps } from './cms-block-missing';
