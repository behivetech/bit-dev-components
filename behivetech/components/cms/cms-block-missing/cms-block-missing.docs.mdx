---
description: 'A CmsBlockMissing component.'
labels: ['Missing Block']
---

import { CmsBlockMissing } from './cms-block-missing';

## React Block Missing Component
Displays a default component if a component is missing from the importedComponents list prop for the CmsArea componpent. This is a dependency of the CmsBlocks component and not needed for any sort of development unless components like CmsBlocks or CmsArea are being customized.
