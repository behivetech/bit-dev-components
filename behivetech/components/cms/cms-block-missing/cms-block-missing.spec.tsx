import * as React from 'react';
import { render } from '@testing-library/react';
import { BasicCmsBlockMissing } from './cms-block-missing.composition';

it('should render with the correct content and attributes', () => {
    const { asFragment } = render(<BasicCmsBlockMissing />);

    expect(asFragment()).toMatchSnapshot();
});
