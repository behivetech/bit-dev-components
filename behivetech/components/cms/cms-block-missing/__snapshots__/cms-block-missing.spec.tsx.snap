// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with the correct content and attributes 1`] = `
<DocumentFragment>
  <div
    class="cms-block-missing"
    role="button"
    style="background-color: rgb(153, 0, 0); color: rgb(255, 255, 255);"
  >
    COMPONENT IMPORT ERROR: The mockMissingComponent Componpent is not available from the importedBlocks prop.
  </div>
</DocumentFragment>
`;
